### PR TITLE
Cross-Correlator Updates

### PR DESCRIPTION
**Pull request by rh-downunder**
## What does this PR implement

Re-ordered cross-correlation steps to conform to Bensen et al. 2007 and updated tests. Made time-tagging optional. Added a first-cut script for pre-processing waveform data to facilitate continental-scale ambient-noise tomography.

### CODE REVIEW

**Comment by medlin01GA** [comment]
The Travis CI test for the PR has failed due to timing out on Python 2.7. Could you please try updating the config for Travis CI to increase allowed time for unit tests?

File: seismic/ASDFdatabase/asdf_preprocess.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    # read inventory
+    inv = None
+    try:
+        inv = read_inventory(instrument_response_inventory)
+    except Exception as e:
+        print (e)
+        raise RuntimeError('Failed to read inventory: %s' % (instrument_response_inventory))
+    # end try
+
+    for s in stations:
```
Comment (medlin01GA) [inline seismic/ASDFdatabase/asdf_preprocess.py]: Please add a comment documenting why garbage collection is needed in this loop. In general it's something one shouldn't have to worry about.

File: seismic/ASDFdatabase/asdf_preprocess.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                inventory_cache[netsta] = inv
+                netstaInv = inv
+            # end if
+        # end if
+    # end if
+
+    return netstaInv, inventory_cache
+# end func
+
+def create_station_asdf(input_asdf, output_folder, resample_rate,
```
Comment (medlin01GA) [inline seismic/ASDFdatabase/asdf_preprocess.py]: Consider replacing print statements with logging in this function.

File: seismic/ASDFdatabase/asdf_preprocess.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+# end func
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('input-asdf', required=True,
+                type=click.Path(exists=True))
+@click.argument('output-folder', required=True,
+                type=click.Path(exists=True))
+@click.option('--resample-rate', default=10,
+              help="Resample rate in Hz; default is 10 Hz")
```
Comment (medlin01GA) [inline seismic/ASDFdatabase/asdf_preprocess.py]: You could also use the `show_default=True` flag to have the `click` CLI documentation display the default value.

File: seismic/xcorqc/correlator.py [hunk 4]
```diff
@@ -354,11 +359,12 @@ def outputConfigParameters():
                                                      "single CC function, aimed at producing empirical Greens "
                                                      "functions for surface wave tomography.")
 @click.option('--restart', default=False, is_flag=True, help='Restart job')
+@click.option('--no-tracking-tag', default=False, is_flag=True, help='Do not tag output file names with a time-tag')
 def main(data_source1, data_source2, output_path, interval_seconds, window_seconds, resample_rate,
```
Comment (medlin01GA) [inline seismic/xcorqc/correlator.py]: At some point I suggest incorporating all these many options into an options class to encapsulate them.

File: seismic/xcorqc/xcorqc.py [hunk 5]
```diff
@@ -695,19 +691,19 @@ def IntervalStackXCorr(refds, tempds,
...
-        sr = root_grp.createVariable('SampleRate', 'f4')
         lon1 = root_grp.createVariable('Lon1', 'f4')
         lat1 = root_grp.createVariable('Lat1', 'f4')
         lon2 = root_grp.createVariable('Lon2', 'f4')
         lat2 = root_grp.createVariable('Lat2', 'f4')
-        xcorrchan = root_grp.createVariable('CorrChannels', 'S1', ('nchar'))
+        distance = root_grp.createVariable('Distance', 'f4')
 
-        sr[:] = resample_rate
         lon1[:] = refds.unique_coordinates[ref_net_sta][0] if len(refds.unique_coordinates[ref_net_sta]) else -999
```
Comment (medlin01GA) [inline seismic/xcorqc/xcorqc.py]: Consider assigning the magic number `-999` to a named identifier.

**Comment by rh-downunder** [comment]
> The Travis CI test for the PR has failed due to timing out on Python 2.7. Could you please try updating the config for Travis CI to increase allowed time for unit tests?

There's a hard limit of 50 mins for public repos. I've updated the tests to operate with a reduced sampling rate, so they run faster.

File: seismic/ASDFdatabase/asdf_preprocess.py (hunk 2 above)
Comment (rh-downunder) [reply seismic/ASDFdatabase/asdf_preprocess.py]: This is a first-cut of the script; I will spend a bit more time to tidy it up in the near future.

File: seismic/xcorqc/correlator.py (hunk 4 above)
Comment (rh-downunder) [reply seismic/xcorqc/correlator.py]: I've been thinking about reading them all from a yaml file..

**Comment by rh-downunder** [comment]
Thanks Andrew, merging now.

**Pull request merged**